UsageEnvironment/Log.cpp

```diff
@@ -77,20 +86,127 @@ FILE* _Log::AcquireFile() {
   } else {
     FILE* logFileHandle = fopen(logFilePath, "a");
     if( logFileHandle == NULL ) {
-      strcpy(logFilePath, "/tmp/live555.log");
+#ifdef LOG_FILE_TMP_DIR
+      strcpy(logFilePath, LOG_FILE_TMP_DIR "/live555.log");
       logFileHandle = fopen(logFilePath, "a");
+#endif
     }
 
     return logFileHandle;
   }
 }
 
+void _Log::RemoveOldBackup() {
+#ifdef LOG_FILE_BACKUP_DIR
+  DIR* backupDir;
+  struct dirent* dirInfo;
+
+  // Look for old backup log file and remove it.
+  backupDir = opendir(LOG_FILE_BACKUP_DIR);
+  if( backupDir ) {
+    while( (dirInfo = readdir(backupDir)) != NULL ) {
+      if( strncmp(dirInfo->d_name, "live555", 7) == 0 ) {
+        char rmBuf[PATH_MAX];
+
+        // Attempt to remove using system call (unlink() and remove() complain /mnt/config is Read-Only)
+        fprintf(stdout, "Attempting to remove old backup log.\n");
+        snprintf(rmBuf, sizeof(rmBuf), "rm " LOG_FILE_BACKUP_DIR "/%s", dirInfo->d_name);
+        system(rmBuf);
+        break;
+      }
+    }
+  }
+  closedir(backupDir);
+#endif
+}
+
+void _Log::BackupLog() {
+#if defined(LOG_FILE_BACKUP_DIR) && defined(LOG_FILE_TMP_DIR)
+  RemoveOldBackup();  // Remove old backup log if it exists (flash memory should never be at risk of getting filled up)
+  fprintf(stdout, "Attempting to backup log.\n");
+
+  // Backup new log file.
+  char newPath[PATH_MAX];
```

Comment:
I couldn't use the rename() function because I was getting errno 18, which means I am attempting to move a file between two file systems, so I have to manually copy it over.